app/ts/background/background.ts

```diff
@@ -55,8 +55,8 @@ async function updateMetadataForSimulation(simulationState: SimulationState, eth
 	}
 }
 
-export const simulateGovernanceContractExecution = async (pendingTransaction: PendingTransaction, ethereum: EthereumClientService) => {
-	const returnError = (text: string) => ({ error: { message: text} })
+export const simulateGovernanceContractExecution = async (pendingTransaction: PendingTransaction, ethereum: EthereumClientService, userAddressBook: UserAddressBook) => {
+	const returnError = (text: string) => ({ error: { type: 'Other' as const, message: text } })
```

Comment:
Recommend a discriminated union if you are going to be returning multiple different types of objects.
```suggestion
	const returnError = (text: string) => ({ success: false, error: { type: 'Other' as const, message: text } })
```
A similar change would need to be made to the happy-path return so it returns `{ success: true, ... }`.

Typescript has special support for discriminated unions (a union of objects where one key on the object is unique for each subtype in the union), and dealing with discriminated unions is generally a better developer experience in TS than dealing with unions that have different shapes but no discriminant.